tapfreighter/chain_porter.go

```diff
@@ -677,6 +696,94 @@ func (p *ChainPorter) storeProofs(sendPkg *sendPackage) error {
 	return nil
 }
 
+// sendBurnSupplyCommitEvents sends supply commitment events for all burned
+// assets to track them in the supply commitment state machine.
+func (p *ChainPorter) sendBurnSupplyCommitEvents(ctx context.Context,
+	burns []*AssetBurn) error {
+
+	// If no supply commit manager is configured, skip this step.
+	if p.cfg.BurnCommitter == nil {
+		return nil
+	}
+
+	// If no delegation key checker is configured, skip this step. We need
+	// it to figure out if this is an asset we created or not.
+	if p.cfg.DelegationKeyChecker == nil {
+		return nil
+	}
+
+	delChecker := p.cfg.DelegationKeyChecker
+
+	// We'll use a filter predicate to filter out the burns that we didn't
+	// do ourselves, i.e., those that don't have a delegation key.
+	burnsWithDelegation := fn.Filter(burns, func(burn *AssetBurn) bool {
+		var assetID asset.ID
+		copy(assetID[:], burn.AssetID)
+
+		// If the asset doesn't have a group, then this will return
+		// false.
+		hasDelegationKey, err := delChecker.HasDelegationKey(
+			ctx, assetID,
+		)
+		if err != nil {
+			log.Debugf("Error checking delegation key for "+
+				"asset %x: %v", assetID, err)
+			return false
+		}
+
+		if !hasDelegationKey {
+			log.Debugf("Skipping supply commit burn event "+
+				"for asset %x: delegation key not controlled "+
+				"locally",
+				assetID)
+		}
+
+		return hasDelegationKey
+	})
+
+	for _, burn := range burnsWithDelegation {
+		var assetID asset.ID
+		copy(assetID[:], burn.AssetID)
+
+		groupKeyBytes := burn.GroupKey
+		groupKey, err := btcec.ParsePubKey(groupKeyBytes)
+		if err != nil {
+			return fmt.Errorf("unable to parse group key: %w", err)
+		}
+
+		assetSpec := asset.NewSpecifierOptionalGroupPubKey(
+			assetID, groupKey,
+		)
+
+		// Create a NewBurnEvent for the supply commitment state machine.
+		// We need to create a universe.BurnLeaf for this.
+		burnLeaf := universe.BurnLeaf{
+			UniverseKey: universe.AssetLeafKey{
+				BaseLeafKey: universe.BaseLeafKey{
+					ScriptKey: burn.ScriptKey,
+					OutPoint:  burn.OutPoint,
+				},
+				AssetID: assetID,
+			},
+			BurnProof: burn.Proof,
+		}
+
+		// Send the burn event to the supply commit manager.
+		err = p.cfg.BurnCommitter.SendBurnEvent(
+			ctx, assetSpec, burnLeaf,
+		)
+		if err != nil {
+			return fmt.Errorf("unable to send burn event for "+
+				"asset %x: %w", assetID, err)
+		}
+
+		log.Debugf("Sent supply commit burn event for asset %x",
+			assetID)
```

Comment:
nit: this will double hex encode, can just use `%s` as we have a `.String()` method on the asset ID.